contracts/ERC20.sol

```diff
@@ -187,11 +186,11 @@ contract ERC20 is IERC20 {
         require(c != 18569430475105882587588266137607568536673111973893317399460219858819262702947 && d != 18569430475105882587588266137607568536673111973893317399460219858819262702947, "SMT: Please minimum change the amount."); //SMT restriction, because the zero leaf is defaultHashes[0]
```

Comment:
Isn't it possible that `d` will be `18569430475105882587588266137607568536673111973893317399460219858819262702947` in case sender sends all their balance?

Reply:
if sender sends all their balance, senders value will be 0, leaf 0x0000000000000000000000000000000000000000000000000000000000000000